spark/src/cgcloud/spark/spark_box.py

```diff
@@ -272,9 +272,11 @@ def __install_spark( self ):
             SPARK_LOCAL_DIRS=self._lazy_mkdir( spark_dir, "local" ),
             JAVA_HOME='/usr/lib/jvm/java-7-oracle',
             SPARK_MASTER_IP='spark-master',
-            HADOOP_CONF_DIR=fmt( "{install_dir}/hadoop/etc/hadoop" ) )
+            HADOOP_CONF_DIR=fmt( "{install_dir}/hadoop/etc/hadoop" ),
+            SPARK_PUBLIC_DNS="$public_name" )
         with remote_open( spark_env_sh_path, use_sudo=True ) as spark_env_sh:
             spark_env_sh.write( '\n' )
+            spark_env_sh.write( 'public_name=$(curl -s http://169.254.169.254/latest/meta-data/public-hostname)\n' )
```

Comment:
Is the indirection via public_name necessary? If `$public_name` works, `$(curl …)` should, too, right?